zbm-builder.sh

```diff
@@ -1,70 +1,145 @@
+     a 'zfsbootmenu_teardown' hook. This command can be invoked multiple
+     times.
 
-  -L Use local './' source tree instead of remote.
-     (Default: upstream master.)
+  -b <path>
+     Use an alternate build directory for the container.
+     (By default, the directory containing 'zbm-builder.sh' is used.)
 
-  -H Do not copy /etc/hostid into image
-     (Has no effect if ./hostid exists)
+  -d Force use of Docker instead of Podman as container management tool.
 
-  -C Do not copy /etc/zfs/zpool.cache into image
-     (Has no effect if ./zpool.cache exists)
+  -h Display this help text.
 
-  -d Force use of docker instead of podman
+  -i <image>
+     Use a different container image or version as build environment.
+     (By default, the official ghcr.io/zbm-dev/zbm-builder:latest is used.)
+
+  -l <path>
+     Use local ZFSBootMenu source tree at <path> instead of remote repository.
+     (By default, a source archive is pulled by the container from GitHub.)
 
   -t <tag>
-     Build the given ZFSBootMenu tag
-     (Default: upstream master)
+     Build a specific ZFSBootMenu release tag (e.g. v1.12.0, v1.10.1, v1.9.0).
+     (By default, the current upstream master is used.)
 
-  -B <tag>
-     Use the given zbm-builder tag to build images
-     (Default: ghcr.io/zbm-dev/zbm-builder:latest)
-EOF
-}
+  -C Do not integrate /etc/zfs/zpool.cache into build image image from host.
+     (If ./zpool.cache exists, this switch will be ignored.)
 
-if command -v podman >/dev/null 2>&1; then
-  PODMAN=podman
-elif command -v docker >/dev/null 2>&1; then
-  PODMAN=docker
-fi
+  -H Do not integrate /etc/hostid into build image from host.
+     (If ./hostid exists, this switch will be ignored.)
 
+EOF
+}
+  
 SKIP_HOSTID=
 SKIP_CACHE=
 
 BUILD_TAG="ghcr.io/zbm-dev/zbm-builder:latest"
+BUILD_DIRECTORY=
 
 BUILD_ARGS=()
-VOLUME_ARGS=("-v" "${PWD}:/build")
+VOLUME_ARGS=()
+
+HOOKS_EARLY_SETUP=()
+HOOKS_SETUP=()
+HOOKS_TEARDOWN=()
+
+if command -v podman >/dev/null 2>&1; then
+  PODMAN="podman"
+elif command -v docker >/dev/null 2>&1; then
+  PODMAN="docker"
+else
+  echo "Error: No suitable container management front-end found."
+  exit 1
+fi
 
-while getopts "hHLCdt:B:" opt; do
+while getopts "1:2:3:b:dhi:l:t:CH" opt; do
   case "${opt}" in
-    L)
-      VOLUME_ARGS+=("-v" "${PWD}:/zbm")
-     ;;
-    H)
-      SKIP_HOSTID="yes"
-      ;;
-    C)
-      SKIP_CACHE="yes"
+    1)
+      sanitise_file "${OPTARG}"
+      HOOKS_EARLY_SETUP+=( "${OPTARG}" )
+    ;;
+    2)
+      sanitise_file "${OPTARG}"
+      HOOKS_SETUP+=( "${OPTARG}" )
+    ;;
+    3)
+      sanitise_file "${OPTARG}"
+      HOOKS_TEARDOWN+=( "${OPTARG}" )
+    ;;
+    b)
+      sanitise_path
+      BUILD_DIRECTORY="${OPTARG}"
       ;;
     d)
       PODMAN=docker
       ;;
+    h)
+      usage
+      exit 0
+      ;;
+    i)
+      BUILD_TAG="${OPTARG}"
+      ;;
+    l)
+      sanitise_path 
+      VOLUME_ARGS+=( "-v" "${OPTARG}:/zbm" )
```

Comment:
As with `-b`, just overwrite the desired local repository.
```suggestion
      LOCAL_REPO="${OPTARG}"
```

Reply:
:+1: